### PR TITLE
Dynamic client with SPI + SPEC update

### PR DESCRIPTION
**Pull request by thejibz**

### CODE REVIEW

File: client/spec/pom.xml [hunk 1]
```diff
@@ -14,29 +14,59 @@
...
+            <groupId>${project.groupId}</groupId>
+            <artifactId>microprofile-graphql-client-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <!-- Needed to generate the class diagram for the client API -->
+        <dependency>
+            <groupId>jakarta.json</groupId>
+            <artifactId>jakarta.json-api</artifactId>
+            <version>1.1.6</version>
```
Comment (phillip-kruger) [inline client/spec/pom.xml:45]: Can this be scope provided ?

File: client/tck/pom.xml [hunk 2]
```diff
@@ -27,13 +27,32 @@
     <name>MicroProfile GraphQL Client :: TCK</name>
     <description>Client-side code-first GraphQL APIs for MicroProfile :: TCK</description>
 
+    <properties>
+        <!-- Testing -->
+        <junit.jupiter.version>5.6.0</junit.jupiter.version>
+        <graphql-java.version>14.0</graphql-java.version>
```
Comment (phillip-kruger) [inline client/tck/pom.xml:33]: Version 15 is out

Comment (thejibz) [reply client/tck/pom.xml:33]: I will update this.

**Comment by thejibz** [comment]
I will try to amend the commits with the signoff

File: client/spec/pom.xml (hunk 1 above)
Comment (thejibz) [reply client/spec/pom.xml:45]: Tested it with "provided" and it's OK. I will update.

**Comment by thejibz** [comment]
I will close this PR and make a new one with a big "signed-off" commit... Too hard to rebase all the commit currently.

**Pull request closed without merging**